modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlQuerySplitter.java

```diff
@@ -1868,7 +1869,13 @@ private void splitAggregate(
 
                     GridSqlElement sumDownRdc = aggregate(false, SUM).addChild(column(cntMapAggAlias));
 
-                    rdcAgg = op(GridSqlOperationType.DIVIDE, sumUpRdc, sumDownRdc);
+                    if (!isFractionalType(agg.resultType().type())) {
+                        sumUpRdc =  function(CAST).resultType(GridSqlType.BIGINT).addChild(sumUpRdc);
+                        sumDownRdc = function(CAST).resultType(GridSqlType.BIGINT).addChild(sumDownRdc);
```

Comment:
No, H2 makes this SUM return `DECIMAL` instead of `BIGINT`. You can check the `DataType.getAddProofType()` and its usages. 

As for the division by 0, I added the test `testEmptyCacheAggregates` for this case. Not sure how, but it works OK for empty resultset.